elijahcole/sinr
# Task fine tuner

Features:
- data extraction scripts
- fine tuning suites
- sbatch scripts to use trainers on unity
- report on decisions made an challenges: https://docs.google.com/document/d/17t-MRulBXyp-WsPVg_WaYRnvEutDUfsqRNVWgjVVFww/edit

Goal:
- Providing a starter point for future research on geomodel fine tuning and annotation integration
- Note limitations discovered during development
- Document the new code

How to test:
- on unity clone the repository to your Personal space, not the pi space, work in your personal space or you have to change the paths
- switch to Task-fine-tuner branch, follow along set up instructions listed in the readme
- once you have pretrained models, and original training data, activate your sinr_icml environment so you can start fine-tuning `python fine_tune_main.py`, an example.csv is provided to act as a sample annotation data.
- You can change parameters, such as lr, in fine_tune_main to observe different effects on training
- Use `python viz_map.py --name fine_tuned --model_path /path/to/inaturalist-sinr/fine-tuned/demo/${YOUR_SAVED_MODEL_NAME}.pt --taxa_id ${TAXA}` to display predictions for a given taxa
- Details about fine tuning steps can be found [here](https://github.com/UMassCDS/inaturalist-sinr/tree/TASK-fine-tuner?tab=readme-ov-file#fine-tuning)

- To test `data_extraction.ipynb`, clone the repo to the computer that is running your database instance and enter database secrets to .env
- Before collecting annotations, you can use iNatAtor app to enter some annotations, if your database is empty you will not get any annotations
- Create the sinr_icml for the new computer
- Select the sinr_icml environment for the `data_extraction.ipynb` and start running the notebook, it will run step by step explaining what is going on
- If the csv file with a datetime has been generated it means it was successful.

If you want to use the generated csv for fine tuning, you have to upload the file to unity using unity dashboard
Details for data extraction can be found [here](https://github.com/UMassCDS/inaturalist-sinr/tree/TASK-fine-tuner?tab=readme-ov-file#fine-tuning)
